protelis/protelis-interpreter/src/main/java/org/protelis/lang/datatype/impl/DeviceUUID.java

```diff
@@ -0,0 +1,60 @@
+package org.protelis.lang.datatype.impl;
+
+import java.util.UUID;
+
+import org.protelis.lang.datatype.DeviceUID;
+
+/** DeviceUID based on UUIDs. */
+public class DeviceUUID implements DeviceUID, Comparable<DeviceUUID> {
+    private static final long serialVersionUID = 1L;
+    private final UUID uid;
+
+    /**
+     * Generate a random UUID.
+     */
+    public DeviceUUID() {
+        this(UUID.randomUUID());
+    }
+
+    /**
+     * @param uid
+     *            the string to use as the UID
+     */
+    public DeviceUUID(final UUID uid) {
+        this.uid = uid;
+    }
+
+    /**
+     * 
+     * @return the underlying string
+     */
+    public UUID getUID() {
+        return uid;
+    }
+
+    @Override
+    public boolean equals(final Object alt) {
+        if (this == alt) {
+            return true;
+        } else if (alt instanceof DeviceUUID) {
```

Comment:
This is fragile in case of subclassing, and may break the commutative semantics of equals. We can either make this class `final` or change the equals.